ocaml/testsuite/tests/typing-layouts-err-msg/value.ml

```diff
@@ -0,0 +1,433 @@
+         of the definition of t_float64 at line 5, characters 0-24.
+       But the layout of t_float64 must be a sublayout of value, because
+         it's an array element.
+|}];;
+
+(* Lazy_expression *)
+let f (x : t_float64) = lazy x
+[%%expect{|
+Line 1, characters 29-30:
+1 | let f (x : t_float64) = lazy x
+                                 ^
+Error: This expression has type t_float64
+       but an expression was expected of type ('a : value)
+       The layout of t_float64 is float64, because
+         of the definition of t_float64 at line 5, characters 0-24.
+       But the layout of t_float64 must be a sublayout of value, because
+         it's a lazy expression.
+|}];;
+
+(* Class_argument *)
+class foo (x : t_float64) =
+  object end;;
+[%%expect{|
+Line 1, characters 10-25:
+1 | class foo (x : t_float64) =
+              ^^^^^^^^^^^^^^^
+Error: This pattern matches values of type t_float64
+       but a pattern was expected which matches values of type ('a : value)
+       The layout of t_float64 is float64, because
+         of the definition of t_float64 at line 5, characters 0-24.
+       But the layout of t_float64 must be a sublayout of value, because
+         it's a term-level argument to a class constructor.
+|}];;
+
+(* Structure_element *)
+module type S = sig val x : t_void end
+[%%expect{|
+Line 1, characters 28-34:
+1 | module type S = sig val x : t_void end
+                                ^^^^^^
+Error: This type signature for x is not a value type.
+       The layout of x is void, because
+         of the definition of t_void at line 6, characters 0-19.
+       But the layout of x must be a sublayout of value, because
+         it's stored in a module structure.
+|}];;
+
+(* Debug_printer_argument *)
+(* See [debug_printer.ml] *)
+
+(* V1_safety_check *)
+type t = {a: t_void; b: int}
+let f (x: t) = match x with | {a; b} -> a
+[%%expect {|
+type t = { a : t_void; b : int; }
+Line 2, characters 6-41:
+2 | let f (x: t) = match x with | {a; b} -> a
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: Non-value detected in [value_kind].
+       Please report this error to the Jane Street compilers team.
+       The layout of t_void is void, because
+         of the definition of t_void at line 6, characters 0-19.
+       But the layout of t_void must be a sublayout of value, because
+         it has to be value for the V1 safety check.
+|}]
+
+(* Captured_in_object *)
+let f (m1 : t_float64) = object
+  val f = fun () -> m1
+end;;
+[%%expect{|
+Line 2, characters 20-22:
+2 |   val f = fun () -> m1
+                        ^^
+Error: m1 must have a type of layout value because it is captured by an object.
+       The layout of t_float64 is float64, because
+         of the definition of t_float64 at line 5, characters 0-24.
+       But the layout of t_float64 must be a sublayout of value, because
+         it's captured in an object.
+|}];;
+
+(* Unknown *)
+(* CR layouts v2.9: add test *)
```

Comment:
If we could add a test, it wouldn't be Unknown! I think we can just skip for now.